src/map/channel.cpp

```diff
@@ -1313,187 +1524,36 @@ unsigned long channel_getColor(const char *color_str) {
-	config_setting_t  *group_list = NULL;
-	int delay = 1000, autojoin = 0, leave = 1, chat = 1, color_override = 0,
-		self_notif = 1, join_notif = 0, leave_notif = 0;
-	int64 type = CHAN_TYPE_PUBLIC;
-	int group_count = 0;
-	const char *name = NULL, *password = NULL, *alias = NULL, *color_str = "Default", *type_str = NULL;
-
-	if (tmp_chan == NULL)
-		return false;
-
-	if (!config_setting_lookup_string(chan, "name", &name)) {
-		ShowError("Please input channel 'name' at '%s' line '%d'! Skipping...\n", chan->file, chan->line);
-		return false;
-	}
-	if (config_setting_lookup_string(chan, "type", &type_str) && !script_get_constant(type_str, &type)) {
-		ShowError("Invalid channel type %s at '%s' line '%d'! Skipping...\n", type_str, chan->file, chan->line);
-		return false;
-	}
-	config_setting_lookup_string(chan, "password", &password);
-	config_setting_lookup_string(chan, "alias", &alias);
-	config_setting_lookup_string(chan, "color", &color_str);
-	config_setting_lookup_int(chan, "delay", &delay);
-	config_setting_lookup_bool(chan, "autojoin", &autojoin);
-	config_setting_lookup_bool(chan, "leave", &leave);
-	config_setting_lookup_bool(chan, "chat", &chat);
-	config_setting_lookup_bool(chan, "color_override", &color_override);
-	config_setting_lookup_bool(chan, "self_notif", &self_notif);
-	config_setting_lookup_bool(chan, "join_notif", &join_notif);
-	config_setting_lookup_bool(chan, "leave_notif", &leave_notif);
-
-	safestrncpy(tmp_chan->name,name+1,sizeof(tmp_chan->name));
-	if (password)
-		safestrncpy(tmp_chan->pass,password,sizeof(tmp_chan->pass));
-	else
-		tmp_chan->pass[0] = '\0';
-	safestrncpy(tmp_chan->alias,alias?alias:name,sizeof(tmp_chan->alias));
-	tmp_chan->msg_delay = delay;
-	tmp_chan->type = (enum Channel_Type)type;
-	tmp_chan->color = channel_getColor(color_str);
-
-	tmp_chan->opt = (autojoin ? CHAN_OPT_AUTOJOIN : 0) |
-		(leave ? CHAN_OPT_CAN_LEAVE : 0) |
-		(chat ? CHAN_OPT_CAN_CHAT : 0) |
-		(color_override ? CHAN_OPT_COLOR_OVERRIDE : 0) |
-		(self_notif ? CHAN_OPT_ANNOUNCE_SELF : 0) |
-		(join_notif ? CHAN_OPT_ANNOUNCE_JOIN : 0) |
-		(leave_notif ? CHAN_OPT_ANNOUNCE_LEAVE : 0);
-
-	if ((group_list = config_setting_get_member(chan, "groupid")) && (group_count = config_setting_length(group_list)) > 0) {
-		int j;
-		CREATE(tmp_chan->groups, unsigned short, group_count);
-		tmp_chan->group_count = group_count;
-		for (j = 0; j < group_count; j++) {
-			int groupid = config_setting_get_int_elem(group_list, j);
-			tmp_chan->groups[j] = groupid;
-		}
-	}
-
-	return true;
-}
-
 /**
  * Read and verify configuration in file
  * Assign table value with value
  */
 void channel_read_config(void) {
-	config_t channels_conf;
-	config_setting_t *chan_setting = NULL;
 
-	if (conf_read_file(&channels_conf, channel_conf)) {
-		ShowError("Cannot read file '%s' for channel connfig.\n", channel_conf);
-		return;
-	}
+	channel_config.private_channel.allow = false;
```

Comment:
Should we assign these during parse if the label isn't defined and using !exist to ensure it's not already parsed, so that they are all in the same place?

Reply:
I don't think we can do that, that only works for public channels like `#main` or `#trade`; not channels that can have multiple of the same like private channels or `#ally`